sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -247,7 +247,11 @@ public Mono<RxDocumentServiceResponse> performRequestInternal(RxDocumentServiceR
             }
 
             Mono<HttpResponse> httpResponseMono = this.httpClient.send(httpRequest, responseTimeout);
-            return toDocumentServiceResponse(httpResponseMono, request, httpRequest);
+
+            MetadataRequestRetryPolicy metadataRequestRetryPolicy = new MetadataRequestRetryPolicy(globalEndpointManager);
+            metadataRequestRetryPolicy.onBeforeSendRequest(request);
+
+            return BackoffRetryUtility.executeRetry(() -> toDocumentServiceResponse(httpResponseMono, request, httpRequest), metadataRequestRetryPolicy);
```

Comment:
I think this policy should use together with WebExceptionRetryPolicy, else SDK will mark region unavailable too aggressive.

when there is a network/connection related issue, SDK will retry few times (controlled in WebExceptionRetryPolicy), and then bubble up the exception to ClientRetryPolicy to mark the region unavailable. 

However with this change, every time we got a connection related issue, it will make the region unavailable, so becomes more aggressive than before.

Reply:
Resolved in next iteration.